Removed vue-responsive-components dependency

onmax opened the pull request:
`vue-responsive-components` is only used in one place and at the same time is ~75KB, seems unnecessary for now.

onmax committed:
Removed vue-responsive-components dependency
- Replaced with new useElementResize composable

sisou commented on src/components/layouts/AddressOverview.vue:
```diff
@@ -219,6 +217,21 @@ export default defineComponent({
+        function maskAddress() {
+            const addressElement = context.refs.$address as HTMLDivElement;
+            const masked = addressElement.clientWidth < addressMaskedWidth.value;
+            addressElement.classList[masked ? 'add' : 'remove']('masked');
```
Please don't use this programmatical way of setting classes, when we use Vue. Vue is reactive, you just set a `boolean` variable and specify the class in the template. When doing it the way you have here, it becomes very hard to see where classes come from.

See also the commit that introduced the `vue-responsive-components` in the first place: https://github.com/nimiq/wallet/commit/e3e6f3fd18ba9cb8baae0477f3c8376bd743670f

Plus, there would be a method for this:
```diff
- addressElement.classList[masked ? 'add' : 'remove']('masked');
+ addressElement.classList.toggle('masked', masked);
```

onmax replied:
You are completely right. I am going to change the PR now. Huge mistake 😑

sisou commented on src/components/layouts/AddressOverview.vue:
```diff
@@ -219,6 +217,21 @@ export default defineComponent({
+        onUnmounted(() => window.removeEventListener('resize', maskAddress));
+
+        function maskAddress() {
+            const addressElement = context.refs.$address as HTMLDivElement;
```
The way to define template refs in the Composition API is with an actual `ref()`:

```js
const $address = ref<HTMLDivElement>(null);
```

Then this becomes:
```suggestion
            const addressElement = $address.value!;
```

sisou commented on src/components/layouts/AddressOverview.vue:
```diff
@@ -219,6 +217,21 @@ export default defineComponent({
+
+            window.addEventListener('resize', maskAddress);
+        });
+        onUnmounted(() => window.removeEventListener('resize', maskAddress));
```
The original commit (https://github.com/nimiq/wallet/commit/e3e6f3fd18ba9cb8baae0477f3c8376bd743670f) that introduced the `vue-responsive-components` dependency used a `ResizeObserver` on the `$address` element instead. While this here with a window resize listener works, I just want you to be aware that it has worse performance, as it fires on _all_ window resizes, whereas a ResizeObserver would only fire when that window resize actually changes the width of the element.

onmax commented:
Given your feedback, I changed the PR. I have added `useElementResize` which only triggers when the element change its size. I am using ResizeObserver which has a 90% market share. So for the 10% remaining I left the `window resize event`.

sisou committed:
Cleanup
